### PR TITLE
added sqlite3 support

### PR DESCRIPTION
**Pull request by fabianbaier**
Run 
```
go run cmd/tsdb/main.go dump ../../snapshots/snapshots/20191017T205052Z-32c83bebdee150eb --label-name=__name__ --label-value=etcd_disk_wal_fsync_duration_seconds_sum --format=sqlite3
```

### CODE REVIEW

File: tsdb/cmd/tsdb/main.go
Comment (fabianbaier) [inline tsdb/cmd/tsdb/main.go]: source-name is probably the wrong name, I think it should be more `dump-output` or so.

Comment (guenter) [reply tsdb/cmd/tsdb/main.go]: We might support other output formats in the future so I would call it `output-file` or similar

File: tsdb/cmd/tsdb/main.go
Comment (guenter) [inline tsdb/cmd/tsdb/main.go:636]: what's the convention for placing typedefs and functions? I would expects typedefs at the top of the file like in C, not sure what's expected in Go

File: tsdb/cmd/tsdb/main.go
Comment (guenter) [inline tsdb/cmd/tsdb/main.go]: should be "error preparing statement", also see below for more instances

File: tsdb/cmd/tsdb/main.go
Comment (guenter) [inline tsdb/cmd/tsdb/main.go:749]: this subquery is probably going to be slow. But you can merge this, we can optimize later if it's too slow.

**Review by guenter — CHANGES_REQUESTED** [review]

File: tsdb/cmd/tsdb/main.go
Comment (fabianbaier) [reply tsdb/cmd/tsdb/main.go:636]: Yeah its a bit different to C in that usually the convention is to keep types close to where they are used. That way its easy for any maintainer to find the types. Now, some core types are typically grouped at the top of a file, but for me `dumpConfiguration` is not so much as a core file as `Configuration` would be. You find a similar pattern with the `writeBenchmark` struct on L159.

**Pull request merged**
